core/contracts/VoteTiming.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title VoteTiming
+ * @dev Computes rounds and phases for an equal length commit-reveal voting cycle. 
+ */
+library VoteTiming {
+    using SafeMath for uint;
+
+    // Note: the phases must be in order. Meaning the first enum value must be the first phase, etc.
+    enum Phase {
+        Commit,
+        Reveal
+    }
+
+    // Note: this MUST match the number of values in the enum above.
+    uint private constant NUM_PHASES = 2;
+
+    struct Data {
+        uint roundId;
+        uint roundStartTime;
+        uint phaseLength;
+    }
+
+    function init(Data storage data, uint phaseLength) internal {
+        data.phaseLength = phaseLength;
+        data.roundId = 1;
+        data.roundStartTime = 0;
+    }
+
+    function getLastUpdatedRoundId(Data storage data) internal view returns (uint) {
+        return data.roundId;
+    }
+
+    function shouldUpdateRoundId(Data storage data, uint currentTime) internal view returns (bool) {
+        return data.roundStartTime != _computeCurrentRoundStartTime(data, currentTime);
+    }
+
+    function updateRoundId(Data storage data, uint currentTime) internal {
+        data.roundId = computeCurrentRoundId(data, currentTime);
+        data.roundStartTime = _computeCurrentRoundStartTime(data, currentTime);
+    }
+
+    function computeCurrentRoundId(Data storage data, uint currentTime) internal view returns (uint roundId) {
+        roundId = data.roundId;
+        if (shouldUpdateRoundId(data, currentTime)) {
+            roundId = roundId.add(1);
```

Comment:
The idea is that rounds should only ever increment by `1`. So if there is no vote for 6 months, the vote round still increments by `1` when a new one occurs. Otherwise, we could statelessly compute the round ID from the time.

Reply:
Doesn't that assumption contradict `computeEstimatedRoundEndTime` ? Looks like that function assumes no skips to get the time.